app/assets/javascripts/controllers/middleware_server/middleware_server_controller.js

```diff
@@ -176,12 +180,11 @@ function MwServerControllerFactory($scope, miqService, mwAddDatasourceService, i
 
   $scope.deleteSelectedDr = function() {
     $document.find('#dr_btn_delete').prop('disabled', true);
-    $document.find('#mw_dr_reports').submit();
+    sendDataWithRx({type: 'delete_jdr'});
   };
 
-  $scope.drChecked = function() {
-    var checkedCount = $document.find('#mw_dr_reports input[type=checkbox]:checked').length;
-    $document.find('#dr_btn_delete').prop('disabled', checkedCount === 0);
+  $scope.drChecked = function(payload) {
+    $document.find('#dr_btn_delete').prop('disabled', payload.countSelected === 0);
```

Comment:
I would highly discourage mutating the DOM from inside an angular controller.
http://ng-learn.org/2014/01/Dom-Manipulations/
I see that the DOM manipulations were added in the previous PR:
https://github.com/ManageIQ/manageiq-ui-classic/pull/1764/files
Starting here: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/assets/javascripts/controllers/middleware_server/middleware_server_controller.js#L177

Reply:
I don't like the DOM manipulation as well, however those buttons are outside angular's scope, so there is no way to enable/disable them from angular, hence the DOM queries.